### ISSUE TITLE
Waypoint error on console when approching Deatholme 

### ISSUE THREAD

**Issue by acidion-ad** [azerothcore/azerothcore-wotlk#17876]
### Current Behaviour

When approaching Deatholme the server will error when loading a specific [creature](https://www.wowhead.com/wotlk/npc=16320/eye-of-darkhan). Same entry every time. 
 
![image](https://github.com/azerothcore/azerothcore-wotlk/assets/85262665/4f13222f-eb69-4b39-a9c1-bb8e9053ccbe)

_WaypointMovementGenerator::LoadPath: creature Eye of Dar'Khan (GUID Full: 0xf130003fc00004dc Type: Creature Entry: 16320  Low: 1244) doesn't have waypoint path id: 0_

### Expected Blizzlike Behaviour

Server shouldn't be throwing errors

### Steps to reproduce the problem

* Approach Deatholme (southern Eversong Woods)
* Monitor server console for errors
* If the area has been loaded recently log out at the gates, restart worldserver and log back in

### AC rev. hash/commit

d3bef76

### Operating system

Windows 11

### Custom changes or Modules

AutoBalance

**Comment by acidion-ad** [comment]
Is anyone able to confirm this issue?

**Comment by evrydayjunglist** [comment]
If you use the worldserver.conf option PreloadAllNonInstancedMapGrids set to 1 you will see this error every time grids are loaded for map 530.

**Comment by sudlud** [comment]
can also (still) confirm on 4cfad0ffa
PreloadAllNonInstancedMapGrids = 1
`WaypointMovementGenerator::LoadPath: creature Eye of Dar'Khan (GUID Full: 0xf130003fc0006257 Type: Creature Entry: 16320  Low: 25175) doesn't have waypoint path id: 0`

**Comment by brunolopes1996** [comment]
can confirm

**Comment by AreChen** [comment]
can confirm

**Comment by Goretantath** [comment]
My errors a bit diff but still the same creature, have preload all non instanced on as-well. 
"SmartScript::ProcessAction: Entry 18945 SourceType 0, Event 3, Link Event 4 not found or invalid, skipped.
WaypointMovementGenerator::LoadPath: creature Eye of Dar'Khan (GUID Full: 0xf130003fc00062a0 Type: Creature Entry: 16320
  Low: 25248) doesn't have waypoint path id: 0"

**Comment by adam8014** [comment]
I have the same problem. So has this problem been resolved?

WaypointMovementGenerator::LoadPath: creature Eye of Dar'Khan (GUID Full: 0xf130003fc0000d73 Type: Creature Entry: 16320 Low: 3443) doesn't have waypoint path id: 0 SmartScript::ProcessAction: Entry 18945 SourceType 0, Event 3, Link Event 4 not found or invalid, skipped.

**Comment by AldebaraanMKII** [comment]
Still happening on commit 77e497a

**Comment by Grimfeather** [comment]
I believe I found the solution.

there are 3 Eye of Dar'Khan creatures with movement type 2. which is waypoints.
82767
82897
82924

but only 2 of those have waypoints in waypoint_data
82897 has no waypoint data.

this can be fixed by either giving the creature waypoints or by changing his movement type to 0 or 1.
```
UPDATE `creature` SET `MovementType` = 0 WHERE `guid` = 82897;
```

this will get rid of the error, but I think the creature should be given waypoints.

--

Entry 18945 SourceType 0, Event 3, Link Event 4 not found or invalid, skipped.

this is not related to it. but I think this can be fixed with:
```
UPDATE `smart_scripts` SET `event_type` = 61 WHERE `entryorguid` = 18945 AND `id` = 4;
```

event 3 is linked to 4. but 4 is not set to 61. which is SMART_EVENT_LINK
https://www.azerothcore.org/wiki/smart_scripts#eventtype

**Comment by AldebaraanMKII** [comment]
Still happening on commit a890f55b5506bfcb4b5a70ec9da970c1833113d2
```
WaypointMovementGenerator::LoadPath: creature Eye of Dar'Khan (GUID Full: 0xf130003fc0004baa Type: Creature Entry: 16320  Low: 19370) doesn't have waypoint path id: 0
```
This error happenned when i was in the valley of trials, no players or playerbots in the server.

also:
```
WaypointMovementGenerator::LoadPath: creature Arzeth the Merciless (GUID Full: 0xf130004b9a005299 Type: Creature Entry: 19354  Low: 21145) doesn't have waypoint path id: 0
```

**Comment by weirdouncle** [comment]
> Still happening on commit [a890f55](https://github.com/azerothcore/azerothcore-wotlk/commit/a890f55b5506bfcb4b5a70ec9da970c1833113d2)
> 
> ```
> WaypointMovementGenerator::LoadPath: creature Eye of Dar'Khan (GUID Full: 0xf130003fc0004baa Type: Creature Entry: 16320  Low: 19370) doesn't have waypoint path id: 0
> ```
> 
> This error happenned when i was in the valley of trials, no players or playerbots in the server.
> 
> also:
> 
> ```
> WaypointMovementGenerator::LoadPath: creature Arzeth the Merciless (GUID Full: 0xf130004b9a005299 Type: Creature Entry: 19354  Low: 21145) doesn't have waypoint path id: 0
> ```

Arzeth the Merciless: The smart script with ID 7 runs only once.  When waypoint finish it looks for the path id from DB which is 0.
This can be fixed by setting this action loop:
`update `smart_scripts` set action_param2 = 1 where entryorguid = 19354 and id = 7`

**Comment by ide1234000** [comment]
Try these：

```
USE acore_world;

-- Complete waypoint fix for Eye of Dar'Khan guid 82897

UPDATE `creature`
SET `position_x` = 6437.28,
    `position_y` = -6360.66,
    `position_z` = 41.276,
    `orientation` = 1.2629,
    `wander_distance` = 0,
    `MovementType` = 2
WHERE `guid` = 82897
  AND `id` = 16320;

DELETE FROM `creature_addon`
WHERE `guid` = 82897;

INSERT INTO `creature_addon` (`guid`, `path_id`, `bytes2`)
VALUES
(82897, 828970, 1);

DELETE FROM `waypoint_data`
WHERE `id` = 828970;

INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`) VALUES
(828970, 1, 6443.78, -6340.43, 41.291),
(828970, 2, 6459.56, -6345.86, 41.208),
(828970, 3, 6453.16, -6365.24, 33.735),
(828970, 4, 6446.68, -6384.37, 41.203),
(828970, 5, 6430.96, -6379.17, 41.291),
(828970, 6, 6437.28, -6360.66, 41.276);
```

**Comment by quasistatic-setup** [comment]
Cross-referencing a detailed analysis I posted in #27258:
https://github.com/azerothcore/azerothcore-wotlk/issues/27258#issuecomment-5687197255

In short:

- On current master, 82897 still has `MovementType = 2` and no `creature_addon` or waypoints.
- #9818 created 82897 as an Eye of Dar'Khan, but its "Pathing for Eye of Dar'Khan" block uses guid 82720 instead. Since #9894, path 827200 has no owner. Its first node is about 1 yd from the 82897 spawn. That suggests, but does not prove, that the block was meant for 82897.
- Please don't apply the SQL from https://github.com/azerothcore/azerothcore-wotlk/issues/17876#issuecomment-4867993166 as it is. It is TrinityCore data (TrinityCore/TrinityCore@1f9f59b4b3) for a different spawn. In AC that position is already used by Wailer 82890, which has the same path, so it would put a second creature on top of it.

#27258 looks like a duplicate of this issue.
